components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -305,6 +305,15 @@ private void sendNotification(String errorResp, String status, String message,
         if (acUrl != null) {
             queryParams += "&" + SAMLSSOConstants.ASSRTN_CONSUMER_URL + "=" + URLEncoder.encode(acUrl, "UTF-8");
         }
+        String relayState;
```

Comment:
Isn't below code has better performance and readability?

String relayState = req.getParameter(SAMLSSOConstants.RELAY_STATE);

// If the request doesn't have a relay state, get it from the session.
        if (StringUtils.isEmpty(relayState)) {
            relayState = sessionDTO.getRelayState();
        }

Reply:
Fixed with 49929bb